lib/src/utils/build_inputs.dart

```diff
@@ -0,0 +1,61 @@
+import 'dart:io';
+
+import 'package:glob/glob.dart';
+import 'package:path/path.dart' as p;
+
+/// Builds and returns the list of inputs on which the formatter should be run.
+///
+/// These inputs are determined by expanding the [include] globs and filtering
+/// out any paths that match the expanded [exclude] globs.
+///
+/// Logs may be output in certain scenarios for debugging purposes.
+///
+/// By default these globs are assumed to be relative to the current working
+/// directory, but that can be overridden via [root] for testing purposes.
+Map<String, dynamic> buildInputs(
+    {List<Glob> exclude, List<Glob> include, String root}) {
+  Map<String, dynamic> returnInputs = {
+    'files': <String>[],
+    'skipped': {
+      'links': <String>[],
+      'excluded': <String>[],
+      'hidden_directories': Set<String>(),
+    },
+  };
+
+  exclude ??= <Glob>[];
+  include ??= <Glob>[];
+  Directory dir = Directory(root ?? '.');
+  for (final entry in dir.listSync(recursive: true, followLinks: false)) {
```

Comment:
Hmm... this got me thinking. Previously, this util would use the list of `include` globs as the starting point for files to format, and then it would filter out any of those files that match one of the `exclude` globs (if there are any).

But, `dartfmt` by default operates on every `.dart` file it can find in the current working directory. Really the only additional functionality we're trying to provide here is a way to exclude certain files, so maybe it would be sufficient to drop the `include` option from the `FormatTool` and only support optional excludes.

In other words, `FormatTool()` would format all dart files by default. If there are certain files that shouldn't be formatted, you'd specify that like so: `FormatTool()..exclude = [Glob('lib/**.g.dart')]`. And that should be sufficient - you'd never need to also tweak the input set since it is already everything.